Framework/Algorithms/src/FitPeaks.cpp

```diff
@@ -1320,6 +1348,17 @@ int FitPeaks::estimatePeakParameters(
   return GOOD;
 }
 
+//----------------------------------------------------------------------------------------------
+/** check whether a peak profile is allowed to observe peak width and set width
+ * @brief isObservablePeakProfile
+ * @param peakprofile
+ * @return
+ */
+bool FitPeaks::isObservablePeakProfile(const std::string &peakprofile) {
+  return std::binary_search(m_observablePeakProfiles.begin(),
```

Comment:
Because it is only a small number of elements, I would imagine it would be equally fast to just do `std::find(...) != m_observablePeakProfiles.end()`.